### ISSUE TITLE
Easy 'Custom Module'

### ISSUE THREAD

**Issue by zackpyle** [beaverbuilder/feature-requests#295]
**Is your feature request related to a problem? Please describe.**
I would like to be able to easily create a custom module to make it easy for clients to edit content AND make it easy for me to create a custom module.

**Describe the solution you'd like**
I would like a backend section under the BB Modules called Custom Modules. In there would exist two sections (or tabs): Fields and Template

In the Fields section it would look a little like ACF's backend. There would be a repeater that would allow you to select various types of inputs (text, image, color, dimensions, select dropdowns, number, url, etc). So you could put in whatever fields you would like to show up in your custom module.

The in the Template section, there would be 3 tabs. This would look a lot like the Custom Layout modal for the Post Module:
- Tab 1 would be html. You would be able to write the html for the custom module here using wpbb shortcodes for the fields you added in the Fields section. Maybe these would be formatted like [wpbb-field name=''] or something. You could also click on the (+) icon and select from the fields that you have specified for that module in the Fields section.
- Tab 2 would be css. You could write module specific css here
- Tab 3 would be js. You could write module specific js here

Then after you are done saving your custom module, this module would show up in the BB frontend under a new module section called Custom Modules

**Describe alternatives you've considered**
The two main alternative here are
1. Creating a custom module which is time consuming and you need some php skills to complete https://www.wpbeaverbuilder.com/custom-module-beaver-builder/
2. [Tangible Blocks](https://tangibleplugins.com/products/tangible-blocks/) has basically created something very similar to this, but native would of course be better. And also they are in beta until probably next summer.

**Comment by zackpyle** [comment]
@codente Lots of support here and [in the group](https://www.facebook.com/groups/beaverbuilders/posts/2334100933415909). Has the BB team thought about this feature before? If not, reactions to this idea? Feasibility?

**Comment by termageddon-hans** [comment]
Very cool idea; could make web designer's lives a lot easier too.  Thoughts on opening up ability to save a module and offer it to others to access/activate within this feature?  Even if you could save these as templates and load them up into other sites (my thought being that web designers may have sets of very similarly built sites and may want to use a module from another website). 

.. this could maybe expand to even being able to search pre-built modules that others have created and offer. May be down the road a bit before you'd consider something like this, but wow that could be a cool direction to go down potentially! Or maybe start with 'vetted' pre-built modules. 

Just some random thoughts here. Seems like a great idea!

**Comment by zackpyle** [comment]
@termageddon-hans Yeah, that is a great next step! Cool iteration on my idea! I don't even see it being much of an issue since all of the fields would be standardized by BB anyway. Maybe they'd integrate the uploadable module library with Assistant (which would give me an excuse to try it out?

I'd see it working a bit like ACF's import/export probably?
![Screen Shot 2022-12-24 at 12 16 27](https://user-images.githubusercontent.com/19413506/209447297-164107b4-da5d-4a3c-941c-03586d7ba5c0.jpg)

**Comment by termageddon-hans** [comment]
@zackpyle boom that's it! Just in general this is such a great idea!
